File: src/main/java/com/example/demo/Entity/CarMaster.java
package com.example.demo.Entity;

import java.sql.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "car_master")
@Data
public class CarMaster {
	
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "car_id")
    private int carId;
	
	@ManyToOne
    @JoinColumn(name = "cartype_id")
    private CarTypeMaster carType;
	
	@Column(name = "car_name")
    private String carName;

    @Column(name = "number_plate", unique = true, length = 50)
    private String numberPlate;

    @Column(name = "status")
    private String Status;

    @Column(name = "mileage")
    private double mileage;

    @ManyToOne
    @JoinColumn(name = "hub_id")
    private HubMaster hub;

    @Enumerated(EnumType.STRING)
    @Column(name = "is_available")
    private AvailabilityStatus isAvailable;

    @Column(name = "maintenance_due_date")
    private Date maintenanceDueDate;

    public int getCarId() {
		return carId;
	}

	public void setCarId(int carId) {
		this.carId = carId;
	}

	public CarTypeMaster getCarType() {
		return carType;
	}

	public void setCarType(CarTypeMaster carType) {
		this.carType = carType;
	}

	public String getCarName() {
		return carName;
	}

	public void setCarName(String carName) {
		this.carName = carName;
	}

	public String getNumberPlate() {
		return numberPlate;
	}

	public void setNumberPlate(String numberPlate) {
		this.numberPlate = numberPlate;
	}

	public String getStatus() {
		return Status;
	}

	public void setStatus(String status) {
		Status = status;
	}

	public double getMileage() {
		return mileage;
	}

	public void setMileage(double mileage) {
		this.mileage = mileage;
	}

	public HubMaster getHub() {
		return hub;
	}

	public void setHub(HubMaster hub) {
		this.hub = hub;
	}

	public AvailabilityStatus getIsAvailable() {
		return isAvailable;
	}

	public void setIsAvailable(AvailabilityStatus isAvailable) {
		this.isAvailable = isAvailable;
	}

	public Date getMaintenanceDueDate() {
		return maintenanceDueDate;
	}

	public void setMaintenanceDueDate(Date maintenanceDueDate) {
		this.maintenanceDueDate = maintenanceDueDate;
	}

	public enum AvailabilityStatus {
        Y, N
    }
	
	

}

File: Backend.NET/Models/CarMaster.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FleetManagementSystem.Api.Models;

[Table("car_master")]
public class CarMaster
{
    [Key]
    [Column("car_id")]
    public int CarId { get; set; }

    [ForeignKey("CarType")]
    [Column("cartype_id")]
    public long? CarTypeId { get; set; }
    public CarTypeMaster? CarType { get; set; }

    [Column("car_name")]
    public string? CarName { get; set; }

    [Column("number_plate")]
    public string? NumberPlate { get; set; }

    [Column("status")]
    public string? Status { get; set; }

    [Column("mileage")]
    public double Mileage { get; set; }

    [ForeignKey("Hub")]
    [Column("hub_id")]
    public int? HubId { get; set; }
    public HubMaster? Hub { get; set; }

    [Column("is_available")]
    public string? IsAvailable { get; set; }

    [Column("maintenance_due_date")]
    public DateTime? MaintenanceDueDate { get; set; }

    [Column("image_path")]
    public string? ImagePath { get; set; }

    public bool IsActuallyAvailable()
    {
        return IsAvailable == "Y" || IsAvailable == "YES" || IsAvailable == "True";
    }
}
